testsuite/tests/kuadrant/authorino/metrics/conftest.py

```diff
@@ -0,0 +1,55 @@
+"""Conftest for the Authorino metrics tests"""
+import time
+from importlib import resources
+
+import pytest
+
+from testsuite.openshift.objects.metrics import Prometheus
+
+
+@pytest.fixture(scope="session")
+def prometheus(openshift):
+    """
+    Return an instance of OpenShift metrics client
+    Skip tests if query route is not properly configured
+    """
+    # find thanos-querier route in the openshift-monitoring project
+    # this route allows to query metrics
+    openshift_monitoring = openshift.change_project("openshift-monitoring")
+    routes = openshift_monitoring.routes.for_service("thanos-querier")
+    if len(routes) > 0:
+        url = ("https://" if "tls" in routes[0]["spec"] else "http://") + routes[0]["spec"]["host"]
+        return Prometheus(url, openshift.token, openshift.project)
+
+    return pytest.skip("Skipping metrics tests as query route is not properly configured")
+
+
+@pytest.fixture(scope="module")
+def label_metrics_service(authorino, module_label):
+    """Label Authorino controller-metrics service for the proper discovery"""
+    authorino.metrics_service.label({"app": module_label})
+
+
+# pylint: disable=unused-argument
+@pytest.fixture(scope="module", autouse=True)
+def create_service_monitor(request, openshift, authorino, blame, module_label, prometheus, label_metrics_service):
+    """Create ServiceMonitor object to follow Authorino /metrics and /server-metrics endpoints"""
+    with resources.path("testsuite.resources", "service-monitor.yaml") as path:
+        service_monitor_object = openshift.new_app(
+            path,
+            {
+                "NAME": blame("sm"),
+                "LABEL": module_label,
+            },
+        )
+    request.addfinalizer(service_monitor_object.delete)
+
+    # wait for the new configured endpoints to became available for metrics scraping
+    time.sleep(90)
```

Comment:
Shouldnt just wait for scrape suffice?

Reply:
After a few tests, the wait here appears to be unnecessary at all, since every test now uses its own wait on scrape for the tested metric using backoff. Removed.